server.js

```diff
@@ -0,0 +1,173 @@
+        where: { email: profile.emails[0].value },
+      })
+        .then((user) => {
+        // if a user is found, log them in
+          if (user) {
+            return done(null, user);
+          }
+
+          Users.create({
+            name: profile.displayName,
+            email: profile.emails[0].value,
+          })
+            .then(newUser => done(null, newUser))
+            .catch(err => done(new Error(err)));
+        })
+        .catch((error) => {
+          done(error, null);
+        });
+    });
+  },
+));
+
+app.use(session({
+  secret: process.env.SECRET,
+  key: process.env.KEY,
+  resave: false,
+  saveUninitialized: false,
+  cookie: {
+    expires: false,
+  }, // in milliseconds
+}));
+
+app.use(passport.initialize());
+
+app.use(passport.session());
+
+app.get(
+  '/auth/google',
+  passport.authenticate('google', { scope: ['profile', 'email'] }),
+);
+
+// the callback after google has authenticated the user
+app.get(
+  '/oauth2callback',
+  passport.authenticate('google', {
+    failureRedirect: '/',
+  }),
+  (req, res) => {
+    console.log(req.user.email, req.user.id, 'successfully');
+  },
+);
+
+app.use('/users', isLoggedIn, userRouter);
+app.use('/workoutTypes', typeRouter);
+app.use('/workouts', workoutRouter);
+
+// app.get('*', (req, res) => {
+//   res.sendFile(path.resolve(__dirname, 'src', 'index.html'));
+// })
+
+
+app.listen(port, () => {
+  console.log(`Server is running on port ${port}`);
+});
+
+const records = {};
+
+const job = new CronJob({
+  cronTime: '00 29 21 * * 6',
+  onTick: () => {
+    const today = new Date();
+    const monInMilliSec = new Date().setDate(today.getDate() - 7);
+    const monDate = new Date(monInMilliSec);
+    const fromDate = moment(monDate).format('YYYY-MM-DD HH:mm:ss');
+    const to = moment(today).format('YYYY-MM-DD HH:mm:ss');
+    Workouts.findAll({
+      where: {
+        createdAt: {
+          $between: [fromDate, to],
+        },
+      },
+      attributes: ['userId', [models.sequelize.fn('AVG', models.sequelize.col('count')), 'Average'], 'type'],
+      group: ['userId', 'type'],
+      order: [[models.sequelize.fn('AVG', models.sequelize.col('count')), 'DESC']],
+    })
+      .then((results) => {
+        records.workouts = results.map(result => result.toJSON());
+        const { workouts } = records;
+        const weekData = [];
+
+        workouts.map((workout) => {
+          const findId = _.findIndex(weekData, item =>
+            parseInt(item.userId, 10) === parseInt(workout.userId, 10));
+
+          if (findId !== -1) {
+            const old = weekData[findId];
+            old.average[workout.type] = _.round(parseFloat(workout.Average));
+          } else {
+            weekData.push({
+              userId: workout.userId,
+              average: { [workout.type]: _.round(parseFloat(workout.Average)) },
+            });
+          }
+        });
+        if (weekData) {
+          weekData.map((data, index) => {
```

Comment:
again, while this works, Array.map signature is meant to return a new array, forEach is probably more suitable here

Also, rather than mapping through the list, is it possible to use `bulkCreate` here
And if possible, let us look into using a free logging service, to log errors